internal/authinterceptors/interceptor.go

```diff
@@ -0,0 +1,77 @@
+package authinterceptors
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+)
+
+type userId string
+
+const userValue userId = "jobby"
+
+func GetUserContext(ctx context.Context) string {
+	return string(ctx.Value(userValue).(string))
+}
+
+func NewContext(ctx context.Context, user string) context.Context {
```

Comment:
Nit: a better name will make it more clear what the function is doing.

```suggestion
func WithUser(ctx context.Context, user string) context.Context {
```

Consider the call site:

```
ctx := authinterceptors.WithUser(ctx, user)
```

Is as lot easier to understand than:

```
ctx := authinterceptors.NewContext(ctx, user)
```